portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,144 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+import java.time.DayOfWeek;
+import java.time.LocalDate; 
+
+@WebServlet("/charts")
+public class ChartServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    //Get the day of the week
+    LocalDate localdate = LocalDate.now();
+    DayOfWeek dotw = DayOfWeek.from(localdate);
+    String currentDay = (String)dotw.name(); 
+    response.getWriter().println((String)dotw.name());
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("dayOfWeek");
+    PreparedQuery results = datastore.prepare(query);
+
+    if(results.countEntities() == 0){
+        Entity dayOfWeek = new Entity("dayOfWeek");
+        dayOfWeek.setProperty("dayOfWeek", currentDay);
+        datastore.put(dayOfWeek);
+    }
+
+    //Gets stored date from datastore
+    String storedDate ="";
+    for(Entity entity:results.asIterable()){
+        storedDate = (String)entity.getProperty("dayOfWeek");
+    }
+
+    //If the current day is the same as stored, continue. If not, update stored date
+    if(!(currentDay.equals(storedDate))){
+
+        //Deletes last weeks entries at the start of a new week
+        if(currentDay.equals("MONDAY") && storedDate.equals("SUNDAY")){
+            ArrayList<Key> pastInput = new ArrayList<>();
+            Query newWeekquery = new Query("input");
+            PreparedQuery newWeekresults = datastore.prepare(newWeekquery);
+            Key pastKey = null;
+            for(Entity entity:newWeekresults.asIterable()){
+                pastKey = entity.getKey();
+                pastInput.add(pastKey);
+            }
+            datastore.delete(pastInput);
+        }
+
+        //Changes the stored date to the current date
+        Key storedDateKey = null;
+        try{
+            for(Entity entity:results.asIterable()){
+                storedDateKey = entity.getKey();
+            }
+            Entity today = datastore.get(storedDateKey);
+            today.setProperty("dayOfWeek", currentDay);
+            datastore.delete(storedDateKey);
+            datastore.put(today);
+        }
+        catch(EntityNotFoundException e){
+            System.out.println("Entity not found");
+        }
+
+        
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Please add comments on complex functions like this. Also, consider splitting this up into 2 servlets. I don't really follow why one is POST and the other is GET (you call them both from javascript using fetch)